promql/engine.go

```diff
@@ -719,9 +719,6 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 		// Make the function call.
 		enh.ts = ts
 		result := f(args, enh)
-		if result.ContainsSameLabelset() {
```

Comment:
Having this here was fine

Reply:
When I removed the duplicate label-set check in funcLabelReplace to test ContainsSameLabelset.
```
  testmetric{src="source-value-10",dst="original-destination-value"} 0
  testmetric{src="source-value-20",dst="original-destination-value"} 1

  eval_fail instant at 0m label_replace(testmetric, "src", "", "", "")
```
For above test case  adding this check here was not detecting the error because the metrics were passed to the function in separate iterations.
That's why I moved the check outside the loop https://github.com/prometheus/prometheus/pull/4589/commits/fc9efa414cd03de5634c35505e5712385c305d49#diff-83539908220a826e39b7ba3a4f27d5d6R761